modules/calib3d/src/solvepnp.cpp

```diff
@@ -259,20 +255,20 @@ class PnPRansacCallback CV_FINAL : public PointSetRegistrator::Callback
 
 bool solvePnPRansac(InputArray _opoints, InputArray _ipoints,
                         InputArray _cameraMatrix, InputArray _distCoeffs,
-                        OutputArray _rvec, OutputArray _tvec, bool useExtrinsicGuess,
+                        InputOutputArray _rvec, InputOutputArray _tvec, bool useExtrinsicGuess,
                         int iterationsCount, float reprojectionError, double confidence,
                         OutputArray _inliers, int flags)
 {
     CV_INSTRUMENT_REGION();
 
     Mat opoints0 = _opoints.getMat(), ipoints0 = _ipoints.getMat();
     Mat opoints, ipoints;
-    if( opoints0.depth() == CV_64F || !opoints0.isContinuous() )
```

Comment:
Always use double precision. I don't think there will be performance degradation and this avoid losing precision if inputs are in double.